piquasso/_backends/fock/pure/calculations.py

```diff
@@ -303,3 +306,71 @@ def _add_occupation_number_basis(  # type: ignore
     index = state._space.index(occupation_numbers)
 
     state._state_vector[index] = coefficient
+
+
+def measure_homodyne(
+    state: PureFockState, instruction: Instruction, shots: int
+) -> Result:
+    modes = instruction.modes
+    phi = instruction._all_params["phi"]
+    reduced_state = state.reduced(modes=modes)
+    scaled_hbar = 1 / reduced_state._config.hbar
+
+    if phi != 0:
+        phi_scaled = np.diag(np.exp(1j * np.atleast_1d(phi)))
+        fock_operator = reduced_state._space.get_passive_fock_operator(
+            phi_scaled,
+            modes=instruction.modes,
+            d=reduced_state._space.d,
+            permanent_function=reduced_state._config.permanent_function,
+        )
+        reduced_state._density_matrix = (
+            fock_operator
+            @ reduced_state._density_matrix
+            @ fock_operator.conjugate().transpose()
+        )
+    resolution = 100000
+    cutoff = reduced_state._config.cutoff
+    quadrature_range = np.linspace(-cutoff, cutoff, resolution)
+    scaled_quadrature_range = np.sqrt(1 / state._config.hbar) * quadrature_range
+
+    # Calculating physicist's Hermite polynomials
+
+    hvalues = []
+    for n in range(cutoff):
+        hvalues.append(hermite(scaled_quadrature_range, n))
+
+    hermite_matrix = np.zeros((cutoff, cutoff, resolution))
+    for n, m in product(range(cutoff), repeat=2):
+        hermite_matrix[n][m] = (
+            1.0
+            / np.sqrt(2**n * factorial(n) * 2**m * factorial(m))
+            * hvalues[n]
+            * hvalues[m]
+        )
+    hermite_values = np.expand_dims(reduced_state.density_matrix, -1) * np.expand_dims(
```

Comment:
This backend is for calculations with pure state. Couldn't we make this calculation more efficient knowing that we have a state vector?